sys/sys/uio.h

```diff
@@ -50,15 +50,35 @@ typedef	__off_t	off_t;
 #endif
 
 #ifdef _KERNEL
+/*
+ * The number of inlined iovecs can be measured by looking at the
+ * distribution of M_IOV allocations with dtrace.
+ * The number of inlined iovecs is tuned to capture the majority of
+ * allocations during a kernel build, measured using the dtrace dtmalloc
+ * provider.
+ */
+#define	UIO_INLINE_IOV	2
+/*
+ * Extra UIO flags
+ */
+#define	UIO_EXT_IOVEC	1
 
 struct uio {
 	struct iovec	*uio_iov;	/* scatter/gather list */
+	int	uio_flags;		/* uio iovec buffer flags */
+#define	uio_startcopy uio_iovcnt
 	int	uio_iovcnt;		/* length of scatter/gather list */
 	off_t	uio_offset;		/* offset in target object */
 	ssize_t	uio_resid;		/* remaining bytes to process */
 	enum	uio_seg uio_segflg;	/* address space */
 	enum	uio_rw uio_rw;		/* operation */
 	struct	thread *uio_td;		/* owner */
+#define	uio_endcopy uio_ext_iov
+	union {
+		struct iovec	uio_inline_iov[UIO_INLINE_IOV];	/* inline iovec
+								   array */
+		struct iovec	*uio_ext_iov;	/* external iovec array */
```

Comment:
Above says extra, here says external

Reply:
The `extra` was referring to the flags field. Will change the comment to avoid confusion.